allennlp/training/metrics/attachment_scores.py

```diff
@@ -53,6 +60,12 @@ def __call__(self, # type: ignore
         gold_indices = gold_indices.long()
         gold_labels = gold_labels.long()
 
+        # Multiply by a mask donoting locations of
+        # gold labels which we should ignore.
+        for label in self._ignore_classes:
```

Comment:
Basically I did it the other way first (which uses indices of _dependency labels_ to mask the metric scores) but then I realised they actually do it based on POS tags, so I had to do it in the model as well. I left this here because I could imagine it being useful for research type stuff like "What is my attachment score for only `nsubj?`".